### ISSUE TITLE
lograge doesn't handle 500 or 404 responses

### ISSUE THREAD

**Issue by thoughtafter** [roidrage/lograge#27]
I have an app, which maybe is doing something strange but it seems like 404 errors caused by missing routes and 500 errors get send to the log file as a stack trace not a single line entry. Am I missing something? In the code it certainly looks like it should be handling 500 errors but it doesn't seem to work.

**Comment by bf4** [comment]
I do something like

```ruby
      if event.payload[:exception] && strace = event.payload[:stacktrace]
        quoted_stacktrace = %('#{Array(strace).join("\n\t")}') # depending on if I'm logging as logstash or not
        opts[:stacktrace] = quoted_stacktrace
        # will output as e.g.
        #   "stacktrace":"'[\"/Users/somedev/.rvm/gems/ etc.
        # no need to include the :exception payload as it is already output
        # by the 'error' key, e.g.
        #   "error":"ActionView::Template::Error:bootstrap/carousel.js isn't precompiled"
      end
```


```ruby
require "active_support/notifications/instrumenter"
# https://github.com/rails/rails/blob/4-2-stable/activesupport/lib/active_support/notifications/instrumenter.rb
# also see payload[:stacktrace] in config/application.rb
# below per Carl Lerche to preserve skylight patch and our patch
# https://gist.github.com/carllerche/acd5c34e284d90678ac0
class ActiveSupport::Notifications::Instrumenter
  # :nocov:
  alias_method :instrument_original, :instrument

  # rubocop:disable Lint/RescueException
  def instrument(name, payload = {})
    instrument_original(name, payload) do
      begin
        yield
      rescue Exception => e
        # BF added the below line to include backtrace in lograge
        payload[:stacktrace] = e.backtrace
        raise
      end
    end
  end
  # rubocop:enable Lint/RescueException
  # :nocov:
end
# look into adding an exception object via middleware that we can
# use in the :append_info_to_payload method
# e.g. see
# https://github.com/airbrake/airbrake/blob/master/lib/airbrake/rails/action_controller_catcher.rb#L7
#  and https://github.com/airbrake/airbrake/blob/master/lib/airbrake/rails/error_lookup.rb#L7
# https://github.com/airbrake/airbrake/blob/master/lib/airbrake/rails/middleware.rb#L13
#  and https://github.com/airbrake/airbrake/blob/master/lib/airbrake/railtie.rb
#  and https://github.com/airbrake/airbrake/blob/master/lib/airbrake/rack.rb#L44
# since there is a difference between what is available by a subscriber and a middleware
#  see https://github.com/roidrage/lograge/blob/master/lib/lograge.rb
```

**Comment by chongivan** [comment]
@bf4 , where do you put the above lines of code?

**Comment by aripollak** [comment]
I think this is fixed as of #112?
